### ISSUE TITLE
Errors with mclust // collect plot function

### ISSUE THREAD

**Issue by VincentAlcazer** [jokergoo/cola#169]
Hey,

Thank you very much for this pretty amazing package.

I have an issue precluding the use of both collect plots and cola report function:

I first ran run_all_consensus_partition_methods on an adjusted matrix of n=860 variables / 60 samples. A first error is reported when coming to the CV:mclust method:
"You have an error when doing partition for CV:mclust.
Error in { : 
  task 1 failed - "max(nu, nv) must be strictly less than min(nrow(A), ncol(A))"

I thus run again the function without mclust method:

```
rl = run_all_consensus_partition_methods(cola_mat, 
    top_value_method = c("SD", "MAD", "CV","ATC"),
    partition_method = c("hclust", "kmeans", "pam","skmeans"),
    cores = 8)
```
Everything runs fine but when calling cola_report, all the plots generated using the collect_plots function return an error:
#> Error in {: task 1 failed - "object 'fun_name' not found"

I tried installing the package from both github and bioconductor, it is the same. Any idea on this one?

Thank you

Best,

Vincent

**Comment by VincentAlcazer** [comment]
Here is the rl object 
[rl_object.zip](https://github.com/jokergoo/cola/files/8259927/rl_object.zip)

sessionInfo:

> sessionInfo()
R version 4.1.1 (2021-08-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=French_France.1252  LC_CTYPE=French_France.1252    LC_MONETARY=French_France.1252
[4] LC_NUMERIC=C                   LC_TIME=French_France.1252    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] genefilter_1.76.0     markdown_1.1          knitr_1.37            cola_2.0.0           
 [5] viridis_0.6.2         viridisLite_0.4.0     ggrepel_0.9.1         factoextra_1.0.7     
 [9] ComplexHeatmap_2.10.0 forcats_0.5.1         stringr_1.4.0         dplyr_1.0.7          
[13] purrr_0.3.4           readr_2.0.2           tidyr_1.1.4           tibble_3.1.5         
[17] ggplot2_3.3.5         tidyverse_1.3.1      

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.4.1        circlize_0.4.13        repr_1.1.4            
  [5] splines_4.1.1          GenomeInfoDb_1.30.0    digest_0.6.28          foreach_1.5.1         
  [9] htmltools_0.5.2        magick_2.7.3           fansi_0.5.0            magrittr_2.0.1        
 [13] memoise_2.0.0          cluster_2.1.2          doParallel_1.0.16      tzdb_0.2.0            
 [17] Biostrings_2.62.0      annotate_1.72.0        modelr_0.1.8           matrixStats_0.61.0    
 [21] colorspace_2.0-2       skimr_2.1.3            blob_1.2.2             rvest_1.0.2           
 [25] haven_2.4.3            xfun_0.27              crayon_1.4.2           RCurl_1.98-1.5        
 [29] microbenchmark_1.4.9   jsonlite_1.7.2         impute_1.68.0          brew_1.0-6            
 [33] survival_3.2-11        iterators_1.0.13       glue_1.4.2             polyclip_1.10-0       
 [37] gtable_0.3.0           zlibbioc_1.40.0        XVector_0.34.0         GetoptLong_1.0.5      
 [41] kernlab_0.9-29         shape_1.4.6            prabclus_2.3-2         BiocGenerics_0.40.0   
 [45] DEoptimR_1.0-10        scales_1.1.1           DBI_1.1.1              Rcpp_1.0.7            
 [49] xtable_1.8-4           clue_0.3-60            bit_4.0.4              mclust_5.4.9          
 [53] stats4_4.1.1           httr_1.4.2             RColorBrewer_1.1-2     fpc_2.2-9             
 [57] modeltools_0.2-23      ellipsis_0.3.2         pkgconfig_2.0.3        XML_3.99-0.8          
 [61] flexmix_2.3-17         nnet_7.3-16            dbplyr_2.1.1           utf8_1.2.2            
 [65] tidyselect_1.1.1       rlang_0.4.12           AnnotationDbi_1.56.2   munsell_0.5.0         
 [69] cellranger_1.1.0       tools_4.1.1            cachem_1.0.6           cli_3.1.0             
 [73] generics_0.1.1         RSQLite_2.2.8          broom_0.7.11           evaluate_0.14         
 [77] fastmap_1.1.0          yaml_2.2.1             bit64_4.0.5            fs_1.5.0              
 [81] robustbase_0.93-9      KEGGREST_1.34.0        slam_0.1-50            xml2_1.3.2            
 [85] compiler_4.1.1         rstudioapi_0.13        png_0.1-7              reprex_2.0.1          
 [89] stringi_1.7.5          highr_0.9              lattice_0.20-44        Matrix_1.3-4          
 [93] vctrs_0.3.8            pillar_1.6.5           lifecycle_1.0.1        BiocManager_1.30.16   
 [97] eulerr_6.1.1           GlobalOptions_0.1.2    bitops_1.0-7           irlba_2.3.3           
[101] R6_2.5.1               gridExtra_2.3          IRanges_2.28.0         diceR_1.1.0           
[105] codetools_0.2-18       MASS_7.3-54            assertthat_0.2.1       rjson_0.2.20          
[109] withr_2.4.3            S4Vectors_0.32.0       GenomeInfoDbData_1.2.7 diptest_0.76-0        
[113] parallel_4.1.1         hms_1.1.1              class_7.3-19           rmarkdown_2.11        
[117] skmeans_0.2-14         Biobase_2.54.0         lubridate_1.8.0        base64enc_0.1-3

**Comment by jokergoo** [comment]
I suggest you to update ComplexHeatmap to the newest version from GitHub. The error message implies it might be a inconsistency to the older version of ComplexHeatmap.

Also please update cola from GitHub. I found a small bug and just fixed it.

I ran `cola_report()` on your object `rl` and everything was running smoothly.

<img width="1036" alt="image" src="https://user-images.githubusercontent.com/449218/158555890-6272860c-fd63-4f09-a781-208957b97cfd.png">

**Comment by jokergoo** [comment]
How about restart your R session? because when you install new versions of packages, they won't be reloaded automatically.

**Comment by VincentAlcazer** [comment]
Already done it (I even tried to launch commands from base R instead of Rstudio). I will see if I have the time to manually load the function to better catch the issue.

**Comment by jokergoo** [comment]
Hi everyone, today I realized the error is from the function `collect_plots()` where temporary png files are generated. Currently there is a problem that 1 either some variables (e.g. `fun_name`) cannot be properly sent into the parallel code chunk, and 2 putting `png()` inside the parallel code chunk makes the execution forever. I don't know why and it needs time to fix that.

Current solution is that I replaced the old parallel code with a simple `lapply()`. This means there won't be any parallel execution of generating figures, but at least it is safer.
